### PR TITLE
Alert design layout

### LINKED ISSUE #82: Authentification alerts design
Edit existing html template for alerts and notices and style it with scss

### PR DESCRIPTION
**Pull request by AntonioDannemann**
Created basic layout for alerts, change the divise .yml for personalizing the alert text. other files where changed but turned back to  normal

### CODE REVIEW

**Comment by wJoenn** [comment]
You didn't share a screenshot of your design

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -8,7 +8,7 @@
 @import "config/setup";
 @import "font-awesome";
 @import "aos/dist/aos.css";
-@import "swiper/swiper-bundle";
+// @import "swiper/swiper-bundle";
```
Comment (wJoenn) [inline app/assets/stylesheets/application.scss:11]: What is this ?

File: app/views/devise/shared/_flashes.html.erb [hunk 2]
```diff
@@ -5,7 +5,10 @@
...
 <% if alert %>
-  <div class="alert">
-    <%= alert %>
+  <div class="alert-container" data-controller="hide-devise-alert" data-hide-devise-alert-target="alert">
+    <div class="alert">
+      <p>Ops, something went wrong <i class="fa-regular fa-face-sad-tear"></i></p>
+      <%= alert %>
+    </div>
   </div>
 <% end %>
```
Comment (wJoenn) [inline app/views/devise/shared/_flashes.html.erb]: What about the notices ?

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/views/devise/shared/_flashes.html.erb [hunk 3]
```diff
@@ -5,7 +5,10 @@
...
 
 <% if alert %>
-  <div class="alert">
-    <%= alert %>
+  <div class="alert-container" data-controller="hide-devise-alert" data-hide-devise-alert-target="alert">
+    <div class="alert">
+      <p>Ops, something went wrong <i class="fa-regular fa-face-sad-tear"></i></p>
```
Comment (MarianaCarlota) [inline app/views/devise/shared/_flashes.html.erb]: Please, replace Ops with Oops or Ups

Comment (AntonioDannemann) [reply app/views/devise/shared/_flashes.html.erb]: Ok, leave it to me

File: app/views/devise/shared/_flashes.html.erb (hunk 2 above)
Comment (AntonioDannemann) [reply app/views/devise/shared/_flashes.html.erb]: I will donit soon, i will send the screenshot in a bit

File: app/views/devise/shared/_flashes.html.erb (hunk 3 above)
Comment (wJoenn) [reply app/views/devise/shared/_flashes.html.erb]: Actually I'm not even sure you should be adding any message as you already render `<%= alert %>`

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-26 at 21 43 08
Screenshot 2023-02-26 at 21 42 14

**Comment by wJoenn** [comment]
In my opinion they are not visible enough. They should be bigger and have some form of frame to make it distinguishable from the rest of the website. I'm also really not a fan of the smiley faces in a error message, that doesn't look professional to me and I think these messages should stay as short as possible, only keeping the actual error and nothing else.

Also you can remove the flashes from the login form, they're not needed anymore as they'll be displayed at the top of the screen already.

**Comment by AntonioDannemann** [comment]
Ok, i will make the changes nowCom meus cumprimentos,Antonio DannemannOn 26 Feb 2023, at 21:59, Louis Ramos ***@***.***> wrote:﻿
In my opinion they are not visible enough. They should be bigger and have some form of frame to make it distinguishable from the rest of the website. I'm also really not a fan of the smiley faces in a error message, that doesn't look professional to me and I think these messages should stay as short as possible, only keeping the actual error and nothing else.

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-26 at 22 30 05
Screenshot 2023-02-26 at 22 30 18

**Comment by wJoenn** [comment]
I think the design still needs to be worked on.
My main point of worry is that it blends too much with the rest of the website. It's just gold on black like everything else and doesn't look like an alert or a notice. Also both alerts and notices look the same, they should have a visual aspect that differentiates them and that is immediately understandable by the user like LeWagon's and Bootstrap's alerts are red and notices are green.

It's tucked in the top left corner, it would probably be a good thing to position it as absolute and give it a z-index of 10 so that it's rendered above the rest of the page instead of pushing everything down.

I think the message is still too small too, not that it needs more text, I think you should completely drop the custom messages and just leave the alerts, but the text should be bigger with more padding.

Did you manage to make the alert disappear after a few seconds with Stimulus ?

**Comment by AntonioDannemann** [comment]
I will try this changes now, yeah the stimulus is working, i will try with green and red notifications Com meus cumprimentos,Antonio DannemannOn 26 Feb 2023, at 22:38, Louis Ramos ***@***.***> wrote:﻿
I think the design still needs to be worked on.
My main point of worry is that it blends too much with the rest of the website. It's just gold on black like everything else and doesn't look like an alert or a notice. Also both alerts and notices look the same, they should have a visual aspect that differentiates them and that is immediately understandable by the user like LeWagon's and Bootstrap's alerts are red and notices are green.
It's tucked in the top left corner, it would probably be a good thing to position it with some margin, something like margin: 100px maybe.
I think the message is still too small too, not that it needs more text, I think you should completely drop the custom messages and just leave the alerts, but the text should be bigger with more padding.
Did you manage to make the alert disappear after a few seconds with Stimulus ?

**Comment by wJoenn** [comment]
I know I said green and red but they were example. I don't think green and red would go very well with a dark theme like ours. 🤔 
I think you're just gonna have to try some stuffs until you feel like what you came up with is something you would see on a real website like twitter amazon and whatnot. Try looking for inspiration if you're struggling, midjourney, google, there are plenty of ways to look for ideas.

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-26 at 23 56 55
Screenshot 2023-02-26 at 23 56 08

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-27 at 11 12 49

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-27 at 11 13 21
0/221549539-1a41a599-7d2b-40cc-8939-f2e65c625ed5.png">

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-27 at 11 13 39

**Comment by AntonioDannemann** [comment]
Screenshot 2023-02-27 at 11 13 21

**Comment by AntonioDannemann** [comment]
screenshot_2023-02-27_at_12 04 18_720

**Comment by Glod33** [comment]
Looks awesome!!

Could it be rendered in the middle?

**Comment by AntonioDannemann** [comment]
It can, i think i sent a screenshot yesterday night, with it in the middle, but Louis said that an alert in the middle of the page is to aggressive, than a changed it.On 27 Feb 2023, at 15:17, Rutger Schoone ***@***.***> wrote:﻿
Looks awesome!!
Could it be rendered in the middle?

**Comment by wJoenn** [comment]
It looked like this in the middle
image

I think it's a bit too aggressive, what I mean is this should be a discret alert not something that grabs all your attention while it's there so I suggested to leave it in the corner with some spacing from the left and top

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -7,10 +7,7 @@
 @import "bootstrap/scss/bootstrap";
 @import "config/setup";
 @import "aos/dist/aos.css";
-@import "font-awesome";
-@import "star-rating.js/dist/star-rating.css";
-@import "swiper/swiper-bundle";
-@import "@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder";
+// @import "swiper/swiper-bundle";
```
Comment (wJoenn) [inline app/assets/stylesheets/application.scss]: Can you tell me why you commented this ?

File: app/assets/stylesheets/components/_alert.scss [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  text-align: center;
+  font-size: 22px;
+  border: 1px solid $gold;
+  border-radius: 15px;
+  padding: 10px;
+
+  span{
+    font-size: 20px;
+  }
+}
```
Comment (wJoenn) [inline app/assets/stylesheets/components/_alert.scss]: The `box-sizing` is on by default for every element on the website because we set it up in the setup file so you can remove it.

Can you change the `z-index` to 10 instead please ? 

With position abolute it's better to use `left` and `top` propreties instead of `margin`. Can you remove the margin and add `left: 50px;` instead please ?

background-color should either be `$main-grey` or `$secondary-grey`

File: app/javascript/controllers/hide_devise_alert_controller.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      setTimeout(() => {
+        this.hideAlert();
+      }, 3000);
+    }
+  }
+
+  hideAlert() {
+    this.alertTarget.style.display = "none";
+  }
+}
```
Comment (wJoenn) [inline app/javascript/controllers/hide_devise_alert_controller.js]: You can remove the static targets and use `this.element` instead of `this.alertTarget` `this.hasAlertTarget`

File: app/javascript/controllers/index.js [hunk 7]
```diff
@@ -13,12 +13,12 @@ application.register("aos", AosController)
 import BookingFormPriceController from "./booking_form_price_controller"
 application.register("booking-form-price", BookingFormPriceController)
 
-import Carousel from 'stimulus-carousel'
-application.register('carousel', Carousel)
-
```
Comment (wJoenn) [inline app/javascript/controllers/index.js:18]: Why did you remove this ?

File: app/views/devise/shared/_flashes.html.erb [hunk 8]
```diff
@@ -1,11 +1,17 @@
-<% if notice %>
-  <div>
-    <%= notice %>
-  </div>
-<% end %>
+<div class="devise-alert" data-controller="hide-devise-alert" data-hide-devise-alert-target="alert">
```
Comment (wJoenn) [inline app/views/devise/shared/_flashes.html.erb]: If you use this.element in the controller you can remove the data-target here

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/javascript/controllers/index.js (hunk 7 above)
Comment (AntonioDannemann) [reply app/javascript/controllers/index.js:18]: for some reason when did the `git pull origin master` my controller to hide the alerts went gone, then I did `rails stimulus:manifest:update` on my branch to get it going again

Comment (wJoenn) [reply app/javascript/controllers/index.js:18]: Ok, the Carousel is not part of the manifest so when we update the manifest it's removed from the index.js
You need to add to back manually please

File: app/views/devise/shared/_flashes.html.erb (hunk 8 above)
Comment (AntonioDannemann) [reply app/views/devise/shared/_flashes.html.erb]: will be done on the next commit

File: app/javascript/controllers/hide_devise_alert_controller.js [hunk 9]
```diff
@@ -5,14 +5,14 @@ export default class extends Controller {
   static targets = ["alert"];
```
Comment (wJoenn) [inline app/javascript/controllers/hide_devise_alert_controller.js]: Remove that

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/application.scss (hunk 4 above)
Comment (AntonioDannemann) [reply app/assets/stylesheets/application.scss]: it breaks everything

Comment (AntonioDannemann) [reply app/assets/stylesheets/application.scss]: nevermind, now its working

**Review by wJoenn — APPROVED** [review]
